### PR TITLE
add long-running-test builder

### PR DESCRIPTION
**Pull request by yjbanov**
The corresponding test.dart change is here: https://github.com/flutter/flutter/pull/67324

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -18,3 +90,6 @@ new configuration files by visiting the
 [luci-config web ui](https://luci-config.appspot.com/), logging in, and
```
Comment (christopherfujino) [inline README.md:90]: Could you change this to the full url `https://luci-config.appspot.com/#/projects/flutter`?

File: README.md [hunk 2]
```diff
@@ -6,6 +6,78 @@ Configuration is generated by `lucicfg`, which is part of `depot_tools`. Rather
...
 `config/main.star`. That file can then be run to regenerate the configuration.
 
+## Updating the configuration
+
+- Edit the desired `.star` files (do not edit `.cfg` files; those are generated).
+- Execute `./config/main.star`.
+- Observe that the `.cfg` files have been updated (e.g. via `git status`).
+- Create a pull request containing changes both to the `.star` files and `.cfg` files.
+- Review and submit the pull request as normal.
+- Changes propagate to LUCI automatically. Go to https://luci-config.appspot.com/#/projects/flutter
```
Comment (christopherfujino) [inline README.md:16]: Maybe refer to the more detailed description in "#forcing-the-propagation-of-these-configurations"?

File: README.md [hunk 3]
```diff
@@ -6,6 +6,78 @@ Configuration is generated by `lucicfg`, which is part of `depot_tools`. Rather
...
+called "foo", it has 3 subshards (0, 1, 2_last) and it's currently staged in pull
+request 99999. You can run it using the following `led` command:
+
+```
+led get-builder "luci.flutter.try:Linux web_tests" \
+  | led edit -pa git_ref='refs/pull/99999/head' \
+  | led edit -pa git_url='https://github.com/flutter/flutter' \
+  | led edit -pa shard='foo' \
+  | led edit -pa subshards='["0","1","2_last"]' \
+  | led edit -pa subshard='1' \
```
Comment (christopherfujino) [inline README.md:73]: I think you either want to pass `subshards` if you want to run them all, or pass `subshard` and then override the recipe to be the drone, to run a single subshard.

File: README.md [hunk 4]
```diff
@@ -6,6 +6,78 @@ Configuration is generated by `lucicfg`, which is part of `depot_tools`. Rather
...
+            "dependencies": [{"dependency": "android_sdk"}, {"dependency": "chrome_and_driver"}, {"dependency": "goldctl"}],
+        },
+        caches = [
+            swarming.cache(name = "pub_cache", path = ".pub_cache"),
+            swarming.cache(name = "android_sdk", path = "android29"),
+        ],
+    )
+```
+
+The easiest way to add a new one is to copy an existing builer and tweak the
```
Comment (nturgut) [inline README.md:47]: nit: typo builder.

**Review by nturgut — COMMENTED** [review]
Thanks for adding this, I'll add more comments soon. Still on it.

File: README.md [hunk 5]
```diff
@@ -6,6 +6,78 @@ Configuration is generated by `lucicfg`, which is part of `depot_tools`. Rather
...
+parameters.
+
+### Testing new shards
+
+As of today, it is not possible to test configuration changes until after they
+propagated to LUCI. One way to mitigate the risk of breakage is to run your test
+shard using an existing builder. To do that, find a builder in one of the
+`.star` files (e.g. `framework_config.star`) that satisfies the dependencies of
+the new test shard, then use the `led` tool (bundled with `depot_tools`) to run
+it.
```
Comment (nturgut) [inline README.md:57]: I think "find a builder... that satisfies the dependencies" is not fully accurate. when you run first led command you will see it outputs a huge file, the commands following it alters the output. Since this configuration is very flexible even builder that does not satisfy any dependency is usable by editing right value.

To sum up I'd rephrase this sentence.

File: README.md (hunk 2 above)
Comment (yjbanov) [reply README.md:16]: @christopherfujino Is forcing necessary? Should I say "wait or force" instead?

@nturgut Is "manually add" the same as forced propagation that @christopherfujino is referring to?

File: README.md (hunk 3 above)
Comment (yjbanov) [reply README.md:73]: Cool! Don't need to explain the appending of subshard lists then.

File: README.md [hunk 6]
```diff
@@ -6,6 +6,78 @@ Configuration is generated by `lucicfg`, which is part of `depot_tools`. Rather
...
+- Execute `./config/main.star`.
+- Observe that the `.cfg` files have been updated (e.g. via `git status`).
+- Create a pull request containing changes both to the `.star` files and `.cfg` files.
+- Review and submit the pull request as normal.
+- Changes propagate to LUCI automatically. Go to https://luci-config.appspot.com/#/projects/flutter
+  to check that new changes have been applied.
+
+## Adding new framework test shards
+
+Framework test shards are defined in [dev/bots/test.dart][1]. When you add a new
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply README.md:21]: Is the link to `test.dart` not sufficient to disambiguate? Happy to add docs, just not sure what to write.

File: README.md (hunk 4 above)
Comment (yjbanov) [reply README.md:47]: Done.

File: README.md (hunk 5 above)
Comment (yjbanov) [reply README.md:57]: I don't know `led` enough to write about it though. I'm only writing what I know myself. Happy to copy and paste if you can give me the text. Alternatively, we can add more ways to exercise `led` for testing in follow-up PRs.

File: README.md (hunk 1 above)
Comment (yjbanov) [reply README.md:90]: Done.

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (nturgut) [reply README.md:21]: In the LUCI/Swarming documentation "shard" is used mostly to refer sharding ability that isolate testing provides. I think a small note in the end which indicated shards created with isolated and flutter test shards are different is sufficient.

Thanks for adding these documentations.

File: README.md (hunk 5 above)
Comment (nturgut) [reply README.md:57]: Yes, you are right it's hard to find a good documentation. led --help explains the process of ` get | edit* | launch` which enables us to modify any builder in a way that we can use it for testing our PRs. I personally use the builder with the closest recipe (rather than the closest dependencies) and if I still need to modify the recipe I use: 

```
 ... | led edit-recipe-bundle | led edit -pa git_ref=<SHA_of_recipe_change> | ...
``` 

My suggestion is optional btw.

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (yjbanov) [reply README.md]: Ah, gotcha. Done.

**Review by nturgut — APPROVED** [review]
LGTM. Can you also share a link to example `led` run?

File: README.md (hunk 2 above)
Comment (christopherfujino) [reply README.md]: forcing is not necessary. i guess a lot of times i'm trying to re-schedule a build ASAP. I guess the thing is, if you're waiting to retry, you won't know it has propagated until you click through to that dashboard, and if you've clicked through to the dashboard you might as well update the config. But "wait or force" is correct :)

**Comment by yjbanov** [comment]
Ok, the runs look green after I submitted the new shards over in flutter/flutter:

- https://ci.chromium.org/p/flutter/builders/try/Linux%20SDK%20Drone/91634?
- https://ci.chromium.org/p/flutter/builders/try/Linux%20SDK%20Drone/91635?
- https://ci.chromium.org/p/flutter/builders/try/Linux%20SDK%20Drone/91642?

Merging.
